### PR TITLE
Add modal popup to onboarding cards

### LINKED ISSUE #5345: Onboarding Carousel Popup

### PR DESCRIPTION
**Pull request by Sabreen-Parveen**
Closes #5345 

This is a v2 of onboarding carousel
Whenever a user will tap on the onboarding cards, a modal window will appear instead of redirecting them to respective faqs page.
Currently implemented on `Keep track of your favourite books` card.

###Things to change:
- [x] Add padding in the whole card
- [x] Card should start from first card everytime the onboarding card is clicked"
- [x] The position of the modal (right now it's relative & causing some height issues)
- [x] Tweaking the design and layout of the modal so it's more clear we're seeing a gif/video 
- [x] The z-index of the blur (so it appears below the modal but above the background elements)
- [x] Hide the text "Next" when we're on the last card

### Testing

- Disable caching in openlibrary/plugins/openlibrary/home.py
- Visit home page
- Tap on `Keep track of your favourite books` card

### Screenshot

image

After edit:
image

### Stakeholders

@mekarpeles @RayBB

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/onboardingCard.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            activateDot(curSlide);
+        };
+
+        const init = function () {
+            goToSlide(0);
+            createDots();
+
+            activateDot(0);
+        };
+        init();
```
Comment (RayBB) [inline openlibrary/plugins/openlibrary/js/onboardingCard.js:81]: Why did you decide to put these three calls in the init function but not lines 84-98?

File: openlibrary/plugins/openlibrary/js/onboardingCard.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        // Event handlers
+        btnRight.forEach((btn) => btn.addEventListener('click', nextSlide));
+        document.addEventListener('keydown', function (e) {
+            if (e.key === 'ArrowLeft') prevSlide();
+            e.key === 'ArrowRight' && nextSlide();
```
Comment (RayBB) [inline openlibrary/plugins/openlibrary/js/onboardingCard.js:87]: Would be good to have like 87 in an if just like 86

**Review by RayBB — CHANGES_REQUESTED** [review]
I didn't get two try it yet but screenshot looks nice. Just two small comments on the code here

File: openlibrary/plugins/openlibrary/js/onboardingCard.js (hunk 1 above)
Comment (Sabreen-Parveen) [reply openlibrary/plugins/openlibrary/js/onboardingCard.js:81]: The three calls are for creating the dots which are not created using html, they are created using javascript. They are put together as they are created when the dom is  loaded and script is executed.

**Comment by RayBB** [comment]
@Sabreen-Parveen I'm unable to get the popup to show for keep track of your books locally or when I try to open it in gitpod. Is the latest push working for you?

**Comment by Sabreen-Parveen** [comment]
> @Sabreen-Parveen I'm unable to get the popup to show for keep track of your books locally or when I try to open it in gitpod. Is the latest push working for you?

Yeah, it's working on my system. Did you disable caching in home page, run `make js` and `make css`?

**Comment by RayBB** [comment]
@Sabreen-Parveen it's working now. I forgot to do the make CSS.
It looks pretty great though!

I did notice what I think is a bug.
That is if you use the next button and get to the last item in modal it will hide the next button.
But if you just click on the last dot for the last item the next button shows and when you click it it circles back to the front. 

Here is video showing the issue. 

https://user-images.githubusercontent.com/921217/132725604-9ae3f080-0565-491a-bf4b-5e2fbf9297ba.mp4

File: openlibrary/plugins/openlibrary/js/onboardingCard.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        });
+
+        dotContainer.addEventListener('click', function (e) {
+            if (e.target.classList.contains('dots__dot')) {
+                const { slide } = e.target.dataset;
```
Comment (Sabreen-Parveen) [inline openlibrary/plugins/openlibrary/js/onboardingCard.js:91]: e.target.dataset reutrns a string value

File: openlibrary/plugins/openlibrary/js/onboardingCard.js [hunk 4]
```diff
@@ -47,6 +47,12 @@ export function initOnboardingCard() {
         };
 
         const goToSlide = function (slide) {
+            if (slide === maxSlide-1) {
```
Comment (Sabreen-Parveen) [inline openlibrary/plugins/openlibrary/js/onboardingCard.js:42]: `===` is a strict comparison operate so both operands should be of same type. The value returned by e.target.data is string.

**Comment by Sabreen-Parveen** [comment]
> @Sabreen-Parveen it's working now. I forgot to do the make CSS.
> It looks pretty great though!
> 
> I did notice what I think is a bug.
> That is if you use the next button and get to the last item in modal it will hide the next button.
> But if you just click on the last dot for the last item the next button shows and when you click it it circles back to the front.
> 
> Here is video showing the issue.
> 
>  next.mp4

Well, I guess I found where the problem was, it was due to [comment](https://github.com/internetarchive/openlibrary/pull/5548#discussion_r706131915)

**Comment by RayBB** [comment]
I'm going to close this isn't it hasn't been worked on in a few years but if you'd like to pick it back up feel free to reopen :)

**Pull request closed without merging**